### ISSUE TITLE
accuracy may be overestimated due to using shuffle split

### ISSUE THREAD

**Issue by artdgn** [DefiLlama/yield-ml#1]
Using shuffle split (`train_test_split`) for this time correlated data is probably not a good idea, as it's very likely that the measured accuracy will be overestimated, and you'll think that the model is performing better than it actually is.

Alternatives:
1. For simplicity - just splitting the arrays by index slicing (after sorting by time) without shuffling.
2. Using the specific [time based method from sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html#sklearn.model_selection.TimeSeriesSplit)

A bit more details in [sklearn's docs](https://scikit-learn.org/stable/modules/cross_validation.html#cross-validation-of-time-series-data)

After checking this, it's possible that accuracy will be similar - but you won't know by how much overestimating without checking. 

I would run it myself just out of curiousity, but didn't find instructions to getting the data (maybe worth adding to the readme?)

**Comment by ptuls** [comment]
I too would love access to the data. I agree with @artdgn there might be issues with data leakage in the training here.

**Comment by slasher125** [comment]
thanks both of u. i am aware the setup might have not been the most robust...updated the notebook. splitting off test data  now by time and using time series split instead of vanialla cross validation. Performance is quite similar.

If u want join our discord, i've created a channel there specifically around yield modelling. 

Also, the data is now publicly available here, s3://defillama-datasets/yield-ml/.

Will also launch the latest db snapshot soon which has 3x the amount of data

**Comment by slasher125** [comment]
Closing this one
